lib/Parse/ParsePattern.cpp

```diff
@@ -1333,6 +1333,20 @@ ParserResult<Pattern> Parser::parseMatchingPattern(bool isExprBasic) {
   // matching-pattern ::= expr
   // Fall back to expression parsing for ambiguous forms. Name lookup will
   // disambiguate.
+  if (Tok.is(tok::period_prefix) && peekToken().is(tok::code_complete)) {
```

Comment:
If I understand correctly, I think this doesn't cover `case .foo(.#^HERE^#)`, we don't know the `foo` is an enum element or a function, right? Could you make it work in nested cases too?

Also considering `case .foo(.#^HERE^#)` case, we don't know `foo` is a static function or an enum element. If it's a static function, we can only use `~=` because enum element patterns can't be nested in an expression pattern. Can we differentiate it in `sawSolution()`?

Reply:
`case .foo(.#^HERE^#)` works because the parser correctly parses the `.foo` as an enum element pattern, so we don’t need to worry about it. I’ll add test cases for it.